src/pybind/mgr/cephadm/module.py

```diff
@@ -3173,6 +3174,16 @@ def set_custom_prometheus_alerts(self, alerts_file: str) -> str:
 
     @handle_orch_error
     def set_prometheus_target(self, url: str) -> str:
+        try:
+            if url.startswith("http://") or url.startswith("https://"):
+                return f"Invalid URL '{url}'. It should be in the format host_ip:port"
+            parsed_url_with_scheme = urlparse(f'http://{url}')
+            host = parsed_url_with_scheme.hostname
+            if not host:
+                raise ValueError("Hostname missing.")
+            ipaddress.ip_address(host)
+        except (ValueError, OSError) as e:
```

Comment:
I can't see which operation can raise teh `OSError` in this case.. seems like we are over-capturing exceptions in this case.

Reply:
Removed